dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelector.java

```diff
@@ -63,41 +63,31 @@ public MockInvokersSelector() {
     }
 
     private <T> List<Invoker<T>> getMockedInvokers(final List<Invoker<T>> invokers) {
-        if (!hasMockProviders(invokers)) {
-            return null;
-        }
         List<Invoker<T>> sInvokers = new ArrayList<Invoker<T>>(1);
         for (Invoker<T> invoker : invokers) {
-            if (invoker.getUrl().getProtocol().equals(MOCK_PROTOCOL)) {
+            if (isMockProtocol(invoker)) {
                 sInvokers.add(invoker);
             }
         }
         return sInvokers;
     }
 
     private <T> List<Invoker<T>> getNormalInvokers(final List<Invoker<T>> invokers) {
-        if (!hasMockProviders(invokers)) {
-            return invokers;
```

Comment:
On most occasions, the call will hit this if branch and return directly.  I don't think the new way will bring any performance impromement.

Reply:
> On most occasions, the call will hit this if branch and return directly. I don't think the new way will bring any performance impromement.

There are no obvious differences of performance between the previous way and the new way. However, the simple changes can make the code have better reusability and readability.

Reply:
I worry there'll have a performance drop for allocating Arraylist for every call.

Reply:
You're right, thanks!